metrics/web/lib/project_groups/presentation/widgets/project_group_dialog.dart

```diff
@@ -177,6 +199,38 @@ class _ProjectGroupDialogState extends State<ProjectGroupDialog> {
     return ProjectGroupsStrings.getSelectedCount(selectedProjectIds.length);
   }
 
+  /// A callback that is called when the group name or any of the projects
+  /// in project checkbox list was (un)checked.
+  /// If the group name and the number of selected projects are both valid,
+  /// the [_isActionButtonActive] is set to `true`, otherwise, to `false`.
+  void _handleProjectGroupChanges() {
+    final groupName = _groupNameController?.text ?? '';
+    if (_validateNumberOfSelectedProjects() && _validateGroupName(groupName)) {
+      _isActionButtonActive.value = true;
+    } else {
+      _isActionButtonActive.value = false;
+    }
+  }
+
+  /// Validates the number of selected projects in the project checkbox list.
+  /// The number of selected projects must be <= 20 and > 0 -> ( n ∈ (0; 20] ).
+  bool _validateNumberOfSelectedProjects() {
+    final numberOfSelectedProjects = _projectGroupsNotifier
+        ?.projectGroupDialogViewModel?.selectedProjectIds?.length;
+    if (numberOfSelectedProjects != null &&
+        numberOfSelectedProjects > 0 &&
+        numberOfSelectedProjects <= 20) {
+      return true;
+    }
+    return false;
+  }
+
+  /// Validates the group name field value.
+  /// The name must not be empty.
+  bool _validateGroupName(String groupName) {
+    return groupName.isNotEmpty;
+  }
+
```

Comment:
Validation should be a part of domain - see project_groups/domain/value_objects/project_group_name.dart and project_groups/domain/value_objects/project_group_projects.dart. Also if you have time please check their usages.